--- pfstools-1.8.5/src/pfs/pfs.cpp~	2010-07-14 11:44:27.000000000 +0200
+++ pfstools-1.8.5/src/pfs/pfs.cpp	2013-04-04 21:50:17.637534986 +0200
@@ -525,7 +525,7 @@
   TagList::const_iterator it;
   fprintf( out, "%d" PFSEOL, tags->getSize() );
   for( it = tags->tagsBegin(); it != tags->tagsEnd(); it++ ) {
-    fprintf( out, (const char*)(it->c_str()) );
+    fprintf( out, "%s", (const char*)(it->c_str()) );
     fprintf( out, PFSEOL );
   }
 }
