diff -up pfstools-2.0.4/CMakeLists.txt.octinstall pfstools-2.0.4/CMakeLists.txt
--- pfstools-2.0.4/CMakeLists.txt.octinstall	2015-07-20 13:06:32.533996797 +0200
+++ pfstools-2.0.4/CMakeLists.txt	2015-07-20 13:11:35.728808062 +0200
@@ -277,8 +277,6 @@ OPTION(WITH_HDRVC "Compile HDRVC (intern
 # ======== MKOCTFILE/OCTAVE ===========
 
 OPTION(WITH_Octave "Compile Octave files" ON)
-SET (OCTAVE_DEST_DIR "${CMAKE_INSTALL_PREFIX}/share/pfstools/octave"
-CACHE PATH  "Copy octave functions to this directory")
 
 if( WITH_Octave )
   
@@ -289,7 +287,8 @@ if( WITH_Octave )
   if(MKOCTFILE)
     message(STATUS "mkoctfile found.")
   endif()
-
+  execute_process(COMMAND octave-config --oct-site-dir OUTPUT_VARIABLE OCTAVE_OCT_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND octave-config --m-site-dir OUTPUT_VARIABLE OCTAVE_M_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif( WITH_Octave )
 
 # ======== OpenCV ===========
diff -up pfstools-2.0.4/src/octave/CMakeLists.txt.octinstall pfstools-2.0.4/src/octave/CMakeLists.txt
--- pfstools-2.0.4/src/octave/CMakeLists.txt.octinstall	2015-07-20 13:03:30.448679088 +0200
+++ pfstools-2.0.4/src/octave/CMakeLists.txt	2015-07-20 13:12:42.873544809 +0200
@@ -40,11 +40,11 @@ endforeach(SRC)
 foreach(SRC ${SRC_OCT}) 
 	install (FILES ${CMAKE_CURRENT_BINARY_DIR}/${SRC}.oct 
 	         PERMISSIONS OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE OWNER_WRITE WORLD_READ GROUP_READ OWNER_READ
-	         DESTINATION ${OCTAVE_DEST_DIR})
+	         DESTINATION "${OCTAVE_OCT_DIR}/pfstools")
 endforeach(SRC)
 
 foreach(SRC ${SRC_M}) 
-	install (FILES ${SRC} DESTINATION ${OCTAVE_DEST_DIR})
+	install (FILES ${SRC} DESTINATION "${OCTAVE_M_DIR}/pfstools")
 endforeach(SRC)
 
 install (FILES ${OCT_SCRIPTS} 
